main.tf

```diff
@@ -6,11 +6,14 @@ locals {
   zone_exists     = module.this.enabled && ! var.zone_enabled
   records_enabled = module.this.enabled && length(var.records) > 0
   zone_id         = local.zone_enabled ? join("", cloudflare_zone.default.*.id) : (local.zone_exists ? lookup(data.cloudflare_zones.default[0].zones[0], "id") : null)
+
+  flatten_records = flatten(var.records)
   records = local.records_enabled ? {
-    for record in flatten(var.records) :
-    format("%s%s",
+    for record in local.flatten_records :
+    format("%s%s%s",
       record.name,
       record.type,
+      index(local.flatten_records, record)
```

Comment:
It's probably obvious but could you explain what the value of the index function would return as an example of a record input to this module?